plugins/romloader/romloader.cpp

```diff
@@ -1427,7 +1445,285 @@ void romloader::call(uint32_t ulNetxAddress, uint32_t ulParameterR0, SWIGLUA_REF
+					}
+					else if( ucPacketTyp==MONITOR_PACKET_TYP_Status )
+					{
+						ptPacketStatus = (MIV3_PACKET_STATUS_T*)m_aucPacketInputBuffer;
+						if( m_sizPacketInputBuffer==sizeof(MIV3_PACKET_STATUS_T) )
+						{
+							/* The netX sent a status code. */
+
+							/* Acknowledge the packet. */
+							send_ack(m_ucMonitorSequence);
+							/* Increase the sequence number. */
+							++m_ucMonitorSequence;
+
+							ucStatus = ptPacketStatus->s.ucStatus;
+							if( ucStatus==MONITOR_STATUS_CallFinished )
+							{
+								fOk = true;
+								break;
+							}
+							else if(ucStatus==MONITOR_STATUS_Ok)
+							{
+								m_ptLog->info("Status: MONITOR_STATUS_Ok");
+								fOk = true;
+								break;
+							}
+							else
+							{
+								/* The netX sent an error. */
+								m_ptLog->error("Status != call_finished received. Status %d.", ucStatus);
+								tResult = TRANSPORTSTATUS_NETX_ERROR;
+							}
+						}
+						else
+						{
+							/* The packet type is set to "status", but the size of the packet does not match a valid status packet. */
+							tResult = TRANSPORTSTATUS_INVALID_PACKET_SIZE;
+						}
+					}
+				}
+
+				if (pcProgressData != NULL)
+				{
+					fIsRunning = callback_string(&tLuaFn, pcProgressData, sizProgressData, lCallbackUserData);
+					if( fIsRunning!=true )
+					{
+						/* Send a cancel request to the device. */
+						cancel_operation();
+
+						MUHKUH_PLUGIN_PUSH_ERROR(tLuaFn.L, "%s(%p): the call was canceled!", m_pcName, this);
+						fOk = false;
+						break;
+					}
+				}
+				i--;
+			}
+		}
+	}
+
+	if( fOk!=true )
+	{
+		MUHKUH_PLUGIN_EXIT_ERROR(tLuaFn.L);
+	}
+}
+
+void romloader::call_hboot(SWIGLUA_REF tLuaFn, long lCallbackUserData)
+{
+	bool fOk;
+	TRANSPORTSTATUS_T tResult;
+	MIV3_PACKET_COMMAND_CALL_T tCallCommand;
+	MIV3_PACKET_CANCEL_CALL_T tCancelCallPacket;
+//	const uint8_t aucCancelBuf[1] = { 0x2b };
+	uint8_t ucStatus;
+	bool fIsRunning;
+	char *pcProgressData;
+	size_t sizProgressData;
+	uint8_t ucPacketTyp;
+	MIV3_PACKET_HEADER_T *ptPacketHeader;
+	MIV3_PACKET_STATUS_T *ptPacketStatus;
+//	int i = 10;
+	bool fPacketStillValid;
+	if( m_fIsConnected==false )
+	{
+		MUHKUH_PLUGIN_PUSH_ERROR(tLuaFn.L, "%s(%p): not connected!", m_pcName, this);
+		fOk = false;
+	}
+	else
+	{	
+		m_ptLog->info("use MONITOR_PACKET_TYP_Command_Start_Hboot");
+		tCallCommand.s.tHeader.s.ucPacketType = MONITOR_PACKET_TYP_Command_Start_Hboot;
+		// tCallCommand.s.ulAddress = HTONETX32(ulNetxAddress); not used for this call. structure is recycled 
+		// tCallCommand.s.ulR0 = HTONETX32(ulParameterR0); not used for this call. structure is recycled 
+		tResult = execute_command(&(tCallCommand.s.tHeader), sizeof(tCallCommand), &fPacketStillValid);
+		if( tResult!=TRANSPORTSTATUS_OK )
+		{
+			MUHKUH_PLUGIN_PUSH_ERROR(tLuaFn.L, "%s(%p): failed to execute command!", m_pcName, this);
+			fOk = false;
+		}
+		else
+		{
+			/* Receive message packets. */
+			while(1)
```

Comment:
The message passing mechanism is not reviewed here, because it's been copied from call()